src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(currentState, action.extraData);
```

Comment:
Instead of using Object.assign directly within the switch case, consider making a copy of the current state and modifying that. This aligns with the checklist item: 'don't mutate object or arrays - it will cause unexpected results later on. You should make copy using Object.assign or spread operator.'